presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergAbstractMetadata.java

```diff
@@ -386,22 +391,28 @@ protected Optional<SystemTable> getIcebergSystemTable(SchemaTableName tableName,
     public ConnectorTableMetadata getTableMetadata(ConnectorSession session, ConnectorTableHandle table)
     {
         IcebergTableHandle icebergTableHandle = (IcebergTableHandle) table;
-        return getTableMetadata(session, icebergTableHandle.getSchemaTableName(), icebergTableHandle.getIcebergTableName());
+        return getTableOrViewMetadata(session, icebergTableHandle.getSchemaTableName(), icebergTableHandle.getIcebergTableName());
     }
 
-    protected ConnectorTableMetadata getTableMetadata(ConnectorSession session, SchemaTableName table, IcebergTableName icebergTableName)
+    protected ConnectorTableMetadata getTableOrViewMetadata(ConnectorSession session, SchemaTableName table, IcebergTableName icebergTableName)
     {
-        Table icebergTable = getIcebergTable(session, new SchemaTableName(table.getSchemaName(), icebergTableName.getTableName()));
-        ImmutableList.Builder<ColumnMetadata> columns = ImmutableList.builder();
-        columns.addAll(getColumnMetadatas(icebergTable));
-        if (icebergTableName.getTableType() == CHANGELOG) {
-            return ChangelogUtil.getChangelogTableMeta(table, typeManager, columns.build());
+        try {
+            Table icebergTable = getIcebergTable(session, new SchemaTableName(table.getSchemaName(), icebergTableName.getTableName()));
+            ImmutableList.Builder<ColumnMetadata> columns = ImmutableList.builder();
+            columns.addAll(getColumnMetadatas(icebergTable));
+            if (icebergTableName.getTableType() == CHANGELOG) {
+                return ChangelogUtil.getChangelogTableMeta(table, typeManager, columns.build());
+            }
+            else {
+                columns.add(PATH_COLUMN_METADATA);
+                columns.add(DATA_SEQUENCE_NUMBER_COLUMN_METADATA);
+            }
+            return new ConnectorTableMetadata(table, columns.build(), createMetadataProperties(icebergTable), getTableComment(icebergTable));
         }
-        else {
-            columns.add(PATH_COLUMN_METADATA);
-            columns.add(DATA_SEQUENCE_NUMBER_COLUMN_METADATA);
+        catch (NoSuchTableException e) {
```

Comment:
Instead of catching this exception, does it make sense to add a new method, `isIcebergView`, then just load the view directly if it's true?

Reply:
Yes, it seems more natural this way. But I did not find an efficient way to implement method `isIcebergView`. `ViewCatalog` in Iceberg library provide a method `boolean viewExists(TableIdentifier identifier)`, based on this method, we can implement `isIcebergView` as follows:

```
boolean isIcebergView(......, SchemaTableName table) {
    ......

    if (catalog instanceof ViewCatalog) {
        return ((ViewCatalog) catalog).viewExists(toIcebergTableIdentifier(table));
    }

    return false;
}
```

However, when we examine `ViewCatalog.viewExists(identifier)` provided by Iceberg library, we find that its implementation is based on `loadView(idenfifier)` as well:

```
boolean viewExists(TableIdentifier identifier) {
    try {
      loadView(identifier);
      return true;
    } catch (NoSuchViewException e) {
      return false;
    }
  }
```

Considering this situation, the above way doesn't seem very efficient, it will do a lot of additional view loading, whether we are querying tables or views.

So the way of capturing exceptions may not seem so natural, but it's an efficient way as far as I can think of. What do you think, which one is more preferable? And open for any other thoughts.

Reply:
Agreed, thanks for the explanation.

Reply:
Can you add some of this information as a comment so folks who read this code later will understand this?

Reply:
Sure, the comment has been added. Please take a look when available, thanks.